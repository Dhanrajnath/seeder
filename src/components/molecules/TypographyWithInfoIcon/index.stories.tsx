import { Meta, StoryObj } from "@storybook/react";
import TypographyWithInfoIcon, { ITypographyWithInfoIconProps } from ".";

export default {
    title: 'Molecules/TypographyInfoIcon',
    component: TypographyWithInfoIcon,
} as Meta;

type TypographyInfoIconType = StoryObj<ITypographyWithInfoIconProps>;


export const TermCap: TypographyInfoIconType = {
    args: {
        title: "Term cap",
        isHeading: false
    }
};

export const CashAccleration: TypographyInfoIconType = {
    args: {
        title: "Cash acceleration",
        isHeading: true
    }
};