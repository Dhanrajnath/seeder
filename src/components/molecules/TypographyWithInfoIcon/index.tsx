import { Box, Tooltip, styled } from "@mui/material";
import Typography from "../../atoms/Typography";
import IconFromSvg from "../../atoms/IconFromSvg";
import theme from "../../../utils/theme";
import InfoSvg from "../../../../public/assets/info.svg";

export interface ITypographyWithInfoIconProps {
    title: string;
    isHeading: boolean;
};

const StyledBox = styled(Box)({
    background: theme.palette.background.default
});

const StyledTypography = styled(Typography)({
    paddingLeft: 1,
    paddingRight: 4,
});

const TypographyWithInfoIcon = ({ title, isHeading }: ITypographyWithInfoIconProps) => {
    return (
        <StyledBox
            display="flex"
            flexDirection="row"
            alignItems="center"
            justifyContent="flex-start"
        >
            <StyledTypography
                variant={isHeading ? "h1" : "body1"}
                children={title}
                color={isHeading ? theme.palette.text.primary : theme.palette.text.secondary}
            />
            <Tooltip title={title} placement="right">
                <span style={{ paddingTop: theme.spacing() }}>
                    <IconFromSvg alt={title} path={InfoSvg} />
                </span>
            </Tooltip>
        </StyledBox>
    );
};

export default TypographyWithInfoIcon;