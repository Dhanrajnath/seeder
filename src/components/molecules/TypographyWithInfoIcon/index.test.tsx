import { fireEvent, render } from "@testing-library/react";
import TypographyWithInfoIcon from ".";

describe('TypographyWihtInfoIcon test', () => {
    it('renders with title and info icon', () => {
        const { container, getByText, getByAltText } = render(<TypographyWithInfoIcon title="Term cap" isHeading={false} />);
        expect(getByText("Term cap")).toBeInTheDocument();
        expect(getByAltText("Term cap")).toBeInTheDocument();
        expect(container.firstChild).toMatchSnapshot();
    });

    it('renders with Heading as title and info icon', () => {
        const { getByText, getByAltText } = render(<TypographyWithInfoIcon title="Cash accleration" isHeading={true} />);
        const headingElement = getByText('Cash accleration');
        expect(headingElement.tagName).toBe('H1');
        expect(getByText("Cash accleration")).toBeInTheDocument();
        expect(getByAltText("Cash accleration")).toBeInTheDocument();
    });

    it('renders tooltip with title and info icon', () => {
        const { getByText } = render(
            <TypographyWithInfoIcon title="Term cap" isHeading={false} />
        );

        const infoIcon = getByText('Term cap');
        fireEvent.mouseOver(infoIcon);

        const tooltip = getByText('Term cap');
        expect(tooltip).toBeInTheDocument();
    });
});