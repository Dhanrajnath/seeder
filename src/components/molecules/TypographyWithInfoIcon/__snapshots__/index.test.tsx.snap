// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TypographyWihtInfoIcon test renders with title and info icon 1`] = `
<div
  class="MuiBox-root css-7dey8k"
>
  <p
    class="MuiTypography-root MuiTypography-body1 css-1jvmuot-MuiTypography-root"
  >
    Term cap
  </p>
  <span
    aria-label="Term cap"
    class=""
    data-mui-internal-clone-element="true"
    style="padding-top: 4px;"
  >
    <img
      alt="Term cap"
      height="inherit"
      src="info.svg"
      width="inherit"
    />
  </span>
</div>
`;
